Mathlib/Algebra/Order/Monoid/LocallyFiniteOrder.lean

```diff
@@ -0,0 +1,205 @@
+          ← card_Ico_zero_add, ha, ← sub_eq_add_neg]
+      · simp [hb', neg_add_eq_sub, eq_sub_iff_add_eq, sub_eq_iff_eq_add,
+          ← Nat.cast_add, ← card_Ico_zero_add, hb, sub_add_eq_add_sub]
+    · have : ¬0 < a + b := by simpa using add_nonpos ha hb
+      simp [ha, hb, card_Ico_zero_add, Ico_eq_empty, this]
+
+variable (G) in
+/-- The canonical embedding (as an ordered monoid hom) from a linearly ordered cancellative
+group into `ℤ`. This is either surjective or zero. -/
+def LocallyFiniteOrder.orderAddMonoidHom :
+    G →+o ℤ where
+  __ := addMonoidHom G
+  monotone' a b hab := by
+    obtain ⟨b, rfl⟩ := add_left_surjective a b
+    replace hab : 0 ≤ b := by simpa using hab
+    suffices 0 ≤ addMonoidHom G b by simpa
+    simp [addMonoidHom, hab]
+
+@[simp]
+lemma LocallyFiniteOrder.orderAddMonoidHom_toAddMonoidHom :
+  orderAddMonoidHom G = addMonoidHom G := rfl
+
+@[simp]
+lemma LocallyFiniteOrder.orderAddMonoidHom_apply (x : G) :
+  orderAddMonoidHom G x = addMonoidHom G x := rfl
+
+lemma LocallyFiniteOrder.orderAddMonoidHom_strictMono :
+    StrictMono (orderAddMonoidHom G) := by
+  rw [strictMono_iff_map_pos]
+  intro g H
+  simpa [addMonoidHom, H.le]
+
+lemma LocallyFiniteOrder.orderAddMonoidHom_bijective [Nontrivial G] :
+    Function.Bijective (orderAddMonoidHom G) := by
+  refine ⟨orderAddMonoidHom_strictMono.injective, ?_⟩
+  suffices 1 ∈ (orderAddMonoidHom G).range by
+    obtain ⟨x, hx⟩ := this
+    exact fun a ↦ ⟨a • x, by simp_all⟩
+  have ⟨a, ha⟩ := exists_zero_lt (α := G)
+  obtain ⟨b, hb⟩ := exists_covBy_of_wellFoundedLT (α := Icc 0 a) (a := ⟨0, by simpa using ha.le⟩)
+    (fun H ↦ ha.not_ge (@H ⟨a, by simpa using ha.le⟩ ha.le))
+  use b.1
+  have : 0 ≤ b.1 := hb.1.le
+  suffices Ico 0 b.1 = {0} by simpa [orderAddMonoidHom, addMonoidHom, this]
+  ext x
+  simp only [mem_Ico, mem_singleton]
+  constructor
+  · rintro ⟨h₁, h₂⟩
+    by_contra hx'
+    have := b.2
+    simp only [Finset.mem_Icc] at this
+    exact hb.2 (c := ⟨x, by simpa [h₁] using h₂.le.trans this.2⟩)
+      (lt_of_le_of_ne h₁ (by simpa using Ne.symm hx')) h₂
+  · rintro rfl
+    simpa using hb.1
+
+variable (G) in
+/-- Any nontrivial linearly ordered abelian group that is locally finite is isomorphic to `ℤ`. -/
+noncomputable
+def LocallyFiniteOrder.orderAddMonoidEquiv [Nontrivial G] :
+    G ≃+o ℤ where
+  __ := orderAddMonoidHom G
+  __ := AddEquiv.ofBijective (orderAddMonoidHom G) orderAddMonoidHom_bijective
+  map_le_map_iff' {a b} := by
+    obtain ⟨b, rfl⟩ := add_left_surjective a b
+    suffices 0 ≤ orderAddMonoidHom G b ↔ 0 ≤ b by simpa
+    obtain hb | hb := le_total 0 b
+    · simp [orderAddMonoidHom, addMonoidHom, hb]
+    · simp [orderAddMonoidHom, addMonoidHom, hb]
+
+lemma LocallyFiniteOrder.orderAddMonoidEquiv_apply [Nontrivial G] (x : G) :
+  orderAddMonoidEquiv G x = addMonoidHom G x := rfl
+
+/-- Any linearly ordered abelian group that is locally finite is embeds to `Multiplicative ℤ`. -/
```

Comment:
here and below in all the docstrings, you have `is embeds`, I think you mean `embeds`